src/lib/FieldType/Validator/TargetContentValidatorInterface.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Core\FieldType\Validator;
+
+use eZ\Publish\Core\FieldType\ValidationError;
+
+interface TargetContentValidatorInterface
+{
+    /**
+     * @param mixed $value
+     */
+    public function validate($value, array $allowedContentTypes = []): ?ValidationError;
```

Comment:
My previous comment applies but if you disagree then as a minimal improvement consider changing `$value` name to something more meaningful i.e. `$contentId` or at least explain it in the type comment.